nio_cli/commands/blockcheck.py

```diff
@@ -0,0 +1,231 @@
+import json
+import os
+import sys
+import subprocess
+
+from .base import Base
+
+
+class BlockCheck(Base):
+
+    # This should be run from inside the block dir
+
+    def __init__(self, options, *args, **kwargs):
+        super().__init__(options, *args, **kwargs)
+        self._block = os.getcwd().split('/')[-1]
+        self.all_contents = os.listdir('.')
+        self.block_files_including_base = [
+            f for f in self.all_contents if self._block in f
+        ]
+        self.block_files = [
+            f for f in self.all_contents if 'block.py' in f and 'base' not in f
+        ]
+        self.blocks_in_spec = []
+        self.spec_versions_dict = {}
+        self.release_versions_dict = {}
+        self.file_versions_and_names_dict = self.get_versions_and_class_names()
+
+    def get_versions_and_class_names(self):
+        name_version_dict = {
+            'classes': [],
+            'versions': [],
+        }
+        for block in self.block_files_including_base:
+            with open(block) as f:
+                lines = [l.rstrip() for l in f.readlines()]
+                for line in lines:
+                    if 'class ' in line:
+                        split1 = line.split(' ')[1]
+                        class_name = split1.split('(')[0]
+                        name_version_dict['classes'].append(class_name)
+                    if '= VersionProperty' in line:
+                        replace1 = line.replace('"', '^')
+                        replace2 = replace1.replace("'", '^')
+                        version_string = replace2.split("^")[1]
+                        # version_string = split1.split("^")[0]
+                        name_version_dict['versions'].append(version_string)
+        return name_version_dict
+
+    def run(self):
+
+        self.print_check('PEP8')
+        self.check_pep8()
+
+        self.print_check('spec.json')
+        self.check_spec()
+
+        self.print_check('README.md')
+        self.check_readme()
+
+        self.print_check('release.json')
+        self.check_release()
+
+        # TODO: Check that block file version is the same as spec/release
+            # open files and look for 'VersionProperty'
+        self.print_check('version')
+        self.check_version()
+
+        # TODO: Check that block file class name is the same as spec/release
+        self.print_check('class and file name')
+        self.check_naming()
+
+    def check_pep8(self):
+
+        shell_pep8 = 'pep8 .'
+        subprocess.call(shell_pep8, shell=True)
+        print('')
+
+    def check_spec(self):
+
+        if os.path.exists('spec.json'):
+            with open('spec.json') as f:
+                spec_dict = json.load(f)
```

Comment:
If the spec.json is empty or invalid json, this prints out a large traceback like:

```
Checking spec.json formatting ...
Traceback (most recent call last):
  File "/Users/dorzel/miniconda3/bin/nio", line 11, in <module>
    load_entry_point('nio-cli', 'console_scripts', 'nio')()
  File "/Users/dorzel/NioCode/nio-cli/nio_cli/cli.py", line 60, in main
    command(options).run()
  File "/Users/dorzel/NioCode/nio-cli/nio_cli/commands/blockcheck.py", line 55, in run
    self.check_spec()
  File "/Users/dorzel/NioCode/nio-cli/nio_cli/commands/blockcheck.py", line 82, in check_spec
    spec_dict = json.load(f)
  File "/Users/dorzel/miniconda3/lib/python3.5/json/__init__.py", line 268, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/Users/dorzel/miniconda3/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/Users/dorzel/miniconda3/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Users/dorzel/miniconda3/lib/python3.5/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
Maybe just grab the last line of that (might just be the exception message) and make a shorter and easier to read notice for the user?